lib/srv/db/cloud/iam.go

```diff
@@ -237,7 +249,9 @@ func (c *IAM) processTask(ctx context.Context, task iamTask) error {
 		Service: c.cfg.AccessPoint,
 		Request: types.AcquireSemaphoreRequest{
 			SemaphoreKind: configurator.cfg.policyName,
-			SemaphoreName: configurator.cfg.identity.GetName(),
+			// Use full identity string as the semaphore name, since two roles
+			// may have the same name in different AWS accounts.
+			SemaphoreName: configurator.cfg.identity.String(),
```

Comment:
@smallinsky the comment explains why I did this, but I just want to make sure this is ok